include/kamping/p2p/sendrecv.hpp

```diff
@@ -0,0 +1,267 @@
+/// - \ref kamping::recv_count() the number of elements to receive. Will be calculated using an additional sendrecv
+/// if not given.
+///
+/// The following parameters are optional:
+/// - \ref kamping::recv_buf() the buffer to receive the message into.  The buffer's underlying
+/// storage must be large enough to hold all received elements. If no \ref kamping::recv_buf() is provided, the \c
+/// value_type of the recv buffer has to be passed as a template parameter to \c sendrecv().
+///
+/// - \ref kamping::send_count() specifying how many elements of the buffer are sent. If omitted, the size of the send
+/// buffer is used as a default.
+///
+/// - \ref kamping::source() receive a message sent from this source rank. Defaults to probing for an arbitrary source,
+/// i.e. \c source(rank::any).
+///
+/// - \ref kamping::send_tag() send message with this tag. Defaults to sending for an arbitrary tag, i.e. \c
+/// tag(tags::any).
+///
+/// - \ref kamping::recv_tag() receive message with this tag. Defaults to receiving for an arbitrary tag, i.e. \c
+/// tag(tags::any).
+///
+/// - \c kamping::status(ignore<>) or \ref kamping::status_out(). Returns info about the received message by setting the
+/// appropriate fields in the status object passed by the user. If \ref kamping::status_out() is passed, constructs a
+/// status object which may be retrieved by the user. The status can be ignored by passing \c
+/// kamping::status(kamping::ignore<>). This is the default.
+///
+/// @tparam Args Automatically deduced template parameters.
+/// @tparam recv_value_type_tparam The type of the message to be received.
+/// @param args All required and any number of the optional parameters described above.
+/// @return Result object wrapping the output parameters to be returned by value.
+///
+/// @see \ref docs/parameter_handling.md for general information about parameter handling in KaMPIng.
+/// <hr>
+
+template <
+    template <typename...>
+    typename DefaultContainerType,
+    template <typename, template <typename...> typename>
+    typename... Plugins>
+template <typename recv_value_type_tparam /* = kamping::internal::unused_tparam */, typename... Args>
+auto kamping::Communicator<DefaultContainerType, Plugins...>::sendrecv(Args... args) const {
+    using namespace kamping::internal;
+    KAMPING_CHECK_PARAMETERS(
+        Args,
+        KAMPING_REQUIRED_PARAMETERS(send_buf, destination),
+        KAMPING_OPTIONAL_PARAMETERS(
+            send_count,
+            send_type,
+            send_tag,
+            recv_buf,
+            recv_tag,
+            source,
+            recv_type,
+            status,
+            recv_count
+        )
+    );
+
+    auto send_buf = internal::select_parameter_type<internal::ParameterType::send_buf>(args...)
+                        .template construct_buffer_or_rebind<UnusedRebindContainer, serialization_support_tag>();
```

Comment:
```suggestion
    auto send_buf = internal::select_parameter_type<internal::ParameterType::send_buf>(args...)
                        .template construct_buffer_or_rebind<UnusedRebindContainer>();
```
We don't support serialization yet, so omit this tag here. When this tag is provided, `send_buf` allows passing serialization buffers, otherwise it is not allowed and a compilation error is issued (which is what we want here)